### PR TITLE
Feature/issue 22

### PR DESCRIPTION
**Pull request by AltusBarry**

### CODE REVIEW

**Comment by AltusBarry** [comment]
Branch adds no new functionality to either api, or app. Everything is contained within tests.
Added functionality for model and serializer to be assigned at runtime via request headers sent from client.
Changes in api or app should be minor and are for the most part only to silence pep8 errors generated by tests.

File: adminapi/api/serializers.py [hunk 1]
```diff
@@ -22,7 +22,9 @@ def create(self, validated_data):
 
     def update(self, instance, validated_data):
         if validated_data.get('username', None) is None:
-            raise serializers.ValidationError({'username': {'This field is required.'}})
+            raise serializers.ValidationError(
+                {'username': {'This field is required.'}}
```
Comment (hedleyroos) [inline adminapi/api/serializers.py]: The second part is not a dict. How does it work?

File: adminapi/tests/__init__.py [hunk 2]
```diff
@@ -370,64 +371,407 @@ def test_user_data_creation_field_error_responses(self):
...
+        # TODO
+        response = self.client.get(
+            reverse("foreign-keys-list"),
+            **{
+                "HTTP_MODEL_CLASS": "ForeignKeyTestModel",
+                "HTTP_SERIALIZER_CLASS": "ForeignModelSerializer"
+            }
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertJSONEqual(
```
Comment (hedleyroos) [inline adminapi/tests/__init__.py:470]: Nice

File: adminapi/tests/__init__.py [hunk 3]
```diff
@@ -10,42 +9,44 @@
...
-class TrivialTest(TestCase):
 
-    @classmethod
-    def setUpClass(cls):
-        cls.obj = TestModel()
-        cls.obj.save()
+class Car(models.Model):
+    title = models.CharField(max_length=100)
+    # Many to one relations
+    foreign_single_relation = models.ForeignKey(Manufacturer)
```
Comment (hedleyroos) [inline adminapi/tests/__init__.py]: Call the key manufacturer

File: adminapi/tests/__init__.py [hunk 4]
```diff
@@ -73,33 +60,33 @@ def tearDownClass(cls):
...
+            reverse("api:login")
         )
         self.assertEqual(response.status_code, 200)
         self.assertJSONEqual(
             response.content,
             {
-                'detail': 'Credentials Validated',
-                'token': self.token.key,
-                'username': self.user.username
+                "detail": "Credentials Validated",
```
Comment (hedleyroos) [inline adminapi/tests/__init__.py]: Make this "Credentials validated" here and in the API. Lowercase is better.

**Comment by hedleyroos** [comment]
+1

**Pull request closed without merging**